### PR TITLE
Add plan enhancements

### PR DESCRIPTION
**Pull request by TY1Fan**
image

Add planned enhancements for:
* Phone number restriction
* Horizontal scroll bar
* Allow multiple classes
* Make add command shorter

### CODE REVIEW

**Comment by Feng1231** [comment]
I think you can provide counter argument for 3rd point that current implementation is good enough for most of our target users, as it is very likely for those students seeking tuition in Singapore to have a Singapore contact number

**Comment by TY1Fan** [comment]
> I think you can provide counter argument for 3rd point that current implementation is good enough for most of our target users, as it is very likely for those students seeking tuition in Singapore to have a Singapore contact number
> 
> Likewise for 4th point, you can add in counter argument that the current implementation is good enough for most of our target users, as it is unlikely for people to have an extremely long name or email.

Good point! Ive made the changes.

**Review by Feng1231 — APPROVED** [review]
lgtm

**Review by Soumyadip-cmd — APPROVED** [review]
lgtm

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -855,3 +855,31 @@ testers are expected to do more *exploratory* testing.
...
+**Team size: 5**
+
+1. **Make Add Command fields shorter:** The current `add command` have 7 compulsory parameters which might be tedious and long even for users who can type fast.
+We plan to make the `add command` shorter by making the `email` field **optional**. This is the **new format:**
+`add n/NAME p/PHONE_NUMBER a/ADDRESS t/SCHEDULE s/SUBJECT r/RATE [e/EMAIL] [paid/PAID_AMOUNT] [owed/OWED_AMOUNT]`, where parameters in square brackets are optional.
+
+1. **Allow students to have multiple classes:** Currently, UGTeach only allow 1 student to have 1 subject and 1 schedule. UGTeach also forbid users from duplicating contacts.
+Hence, users are unable to record multiple classes for students who require tuition for **more than one subject**. 
+Therefore, we plan to combine the `subject` and `schedule` parameters to form a `class` parameter that takes in 1 or more classes (comma-seperated). For instance, the input
+`edit 1 class/Mathematics Monday-1500-1600, Science Wednesday-1200-1400` would mean the first student in UGTeach is receiving tuitions
```
Comment (Feng1231) [inline docs/DeveloperGuide.md:875]: is receiving `tutoring`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -855,3 +855,31 @@ testers are expected to do more *exploratory* testing.
...
+
+**Team size: 5**
+
+1. **Make Add Command fields shorter:** The current `add command` have 7 compulsory parameters which might be tedious and long even for users who can type fast.
+We plan to make the `add command` shorter by making the `email` field **optional**. This is the **new format:**
+`add n/NAME p/PHONE_NUMBER a/ADDRESS t/SCHEDULE s/SUBJECT r/RATE [e/EMAIL] [paid/PAID_AMOUNT] [owed/OWED_AMOUNT]`, where parameters in square brackets are optional.
+
+1. **Allow students to have multiple classes:** Currently, UGTeach only allow 1 student to have 1 subject and 1 schedule. UGTeach also forbid users from duplicating contacts.
+Hence, users are unable to record multiple classes for students who require tuition for **more than one subject**. 
+Therefore, we plan to combine the `subject` and `schedule` parameters to form a `class` parameter that takes in 1 or more classes (comma-seperated). For instance, the input
```
Comment (Feng1231) [inline docs/DeveloperGuide.md:874]: `separated` instead of `seperated`

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -855,3 +855,31 @@ testers are expected to do more *exploratory* testing.
...
+
+<div style="page-break-after: always;"></div>
+
+## **Appendix: Planned Enhancement**
+
+Given below are the planned enhancements for UGTeach (to be implemented in the future).
+
+**Team size: 5**
+
+1. **Make Add Command fields shorter:** The current `add command` have 7 compulsory parameters which might be tedious and long even for users who can type fast.
```
Comment (Feng1231) [inline docs/DeveloperGuide.md:868]: current add command `has`

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -855,3 +855,31 @@ testers are expected to do more *exploratory* testing.
...
+We plan to make the `add command` shorter by making the `email` field **optional**. This is the **new format:**
+`add n/NAME p/PHONE_NUMBER a/ADDRESS t/SCHEDULE s/SUBJECT r/RATE [e/EMAIL] [paid/PAID_AMOUNT] [owed/OWED_AMOUNT]`, where parameters in square brackets are optional.
+
+1. **Allow students to have multiple classes:** Currently, UGTeach only allow 1 student to have 1 subject and 1 schedule. UGTeach also forbid users from duplicating contacts.
+Hence, users are unable to record multiple classes for students who require tuition for **more than one subject**.
```
Comment (Feng1231) [inline docs/DeveloperGuide.md:873]: who require `tutoring`

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -855,3 +855,31 @@ testers are expected to do more *exploratory* testing.
...
+We plan to make the `add command` shorter by making the `email` field **optional**. This is the **new format:**
+`add n/NAME p/PHONE_NUMBER a/ADDRESS t/SCHEDULE s/SUBJECT r/RATE [e/EMAIL] [paid/PAID_AMOUNT] [owed/OWED_AMOUNT]`, where parameters in square brackets are optional.
+
+1. **Allow students to have multiple classes:** Currently, UGTeach only allow 1 student to have 1 subject and 1 schedule. UGTeach also forbid users from duplicating contacts.
+Hence, users are unable to record multiple classes for students who require tuition for **more than one subject**. 
+Therefore, we plan to combine the `subject` and `schedule` parameters to form a `class` parameter that takes in 1 or more classes (comma-seperated). For instance, the input
+`edit 1 class/Mathematics Monday-1500-1600, Science Wednesday-1200-1400` would mean the first student in UGTeach is receiving tuitions
+for Mathematics on Monday (1500-1600) and Science on Wednesday (1200-1400).
+
+1. **Allow phone numbers from other countries:** Currently, UGTeach only allow Singapore phone numbers as we assumed that students (local or international) should have a Singapore number.
```
Comment (Feng1231) [inline docs/DeveloperGuide.md:878]: Currently, ugteach only `allows`

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -855,3 +855,31 @@ testers are expected to do more *exploratory* testing.
...
+`edit 1 class/Mathematics Monday-1500-1600, Science Wednesday-1200-1400` would mean the first student in UGTeach is receiving tuitions
+for Mathematics on Monday (1500-1600) and Science on Wednesday (1200-1400).
+
+1. **Allow phone numbers from other countries:** Currently, UGTeach only allow Singapore phone numbers as we assumed that students (local or international) should have a Singapore number. 
+However, the user might provide tuition to international students who do not have a Singapore number.
+Hence, we plan to **ease the restriction on phone numbers** to allow phone numbers **ranging from 3-digits to 17-digits** since the shortest and longest
+phone number in the world are 3 and 17 digits long respectively, according to the [World Population Review](https://worldpopulationreview.com/country-rankings/phone-number-length-by-country).
+
+1. **Improve UI to be horizontally scrollable**: Currently, UGTeach only allows vertical scrolling as it is unlikely for students to have an extremely long name or email.
+Hence 'extreme' inputs (e.g., name with 1000 characters) are **truncated** which might affect the normal usage of UGTeach.
```
Comment (Feng1231) [inline docs/DeveloperGuide.md:884]: which might `interfere`

**Review by Feng1231 — COMMENTED** [review]
There are some grammar errors according to gpt

**Pull request merged**